### PR TITLE
Updating to use isinstance(str, basestring). using types.StringTypes …

### PR DESCRIPTION
**Pull request by jvlomax**
…is depreciated

### CODE REVIEW

**Comment by jvlomax** [comment]
From the types.py file:

# StringTypes is already outdated.  Instead of writing "type(x) in
# types.StringTypes", you should use "isinstance(x, basestring)".  But
# we keep around for compatibility with Python 2.2.

**Comment by tardyp** [comment]
please remove the 'import types' to make pyflakes happy.

File: slave/buildslave/runprocess.py [hunk 1]
```diff
@@ -443,7 +442,7 @@ def _startCommand(self):
         self.pp = RunProcessPP(self)
 
         self.using_comspec = False
-        if type(self.command) in types.StringTypes:
+        if isinstance(self.command, basestring):
             if runtime.platformType == 'win32':
```
Comment (Stibbons) [inline slave/buildslave/runprocess.py]: are there plan to switch to python 3?

does it make sense to use six?
```
from six import string_types
isinstance(s, string_types)
```

**Comment by jvlomax** [comment]
Yes, this is in preparation to porting to python3. I'n not sure if we want to make buildbot have six as a requirement. a simple solution to make it compatible with both 2 and 3 without using six is to use:
```
try:
  basestring
except NameError:
  basestring = str
```

Not saying it's a better solution. I'm going to ask @djmitche or @jaredgrubb and see what they think before I commit to either pattern

**Comment by bastien31** [comment]
:+1:

**Comment by sa2ajj** [comment]
@jvlomax is this the only place where it's used?

**Comment by jvlomax** [comment]
@sa2ajj It's the only place in buildslave, which is going to be ported before buildmaster

**Comment by sa2ajj** [comment]
Then :+1: for your suggestion. :)

**Comment by jvlomax** [comment]
Since I'm new to the pull request process, I'm wondering if I can submit other patches or push to my fork while this PR is still pending?

**Comment by dblotsky** [comment]
If you push to this branch on your fork, it will show up on this PR. If you push to another branch on your fork, it will not show up on the PR.

**Comment by jvlomax** [comment]
No, but the commits in this PR are also committed on the branch, so they will show up on a PR request from a separate branch as long as the commits are not merged. 

Just want to make sure this is ok, because there will be quite a few PR through summer

**Comment by dblotsky** [comment]
@jvlomax To do what you mean, you will probably want to do you work on (and make PRs from) feature branches on your fork. That will let you keep different sets of work separate from each other and/or keep several PRs in flight at the same time.

**Comment by djmitche** [comment]
If it's easy to avoid requiring six on the buildslave, that will be good.  But it's also no big deal to require it.

As for branches, generally you shouldn't make a pull request for your "master" branch. Instead, for each bit of work that will turn into a pull request, make a new topic branch.  For example, I created https://github.com/djmitche/buildbot-infra/tree/supybot for my work to set up supybot in the buildbot infrastructure repository.  These branches are called "topic branches".

When review takes a while and you have branches depending on other branches, things can get a little tricky, but hopefully things will be reviewed quickly for you.

**Comment by djmitche** [comment]
Another minor note - commit messages should generally use the imperative tense: "update the foo..", 'remove the bar..", etc..

**Comment by djmitche** [comment]
This looks good to me.  I thought "except foo as bar" wasn't supported in Py2.6, but apparently it is!

**Comment by jaredgrubb** [comment]
This looks good to me!

**Comment by sa2ajj** [comment]
@jvlomax Thanks!

**Pull request merged**